api/src/main/java/io/druid/data/input/impl/NewSpatialDimensionSchema.java

```diff
@@ -76,13 +105,24 @@ public boolean equals(Object o)
 
     NewSpatialDimensionSchema that = (NewSpatialDimensionSchema) o;
 
-    return dims != null ? dims.equals(that.dims) : that.dims == null;
+    if (!super.equals(that)) {
```

Comment:
Calling ```super.equals()``` might be an old style, but is still used even in Druid itself (https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/query/select/SelectQuery.java#L236), or Guava (https://github.com/google/guava/blob/master/guava/src/com/google/common/primitives/Longs.java#L678). Maybe we can remove ```super.equals()``` later if we can avoid evaluating equality or do some refactoring. 

I think replacing subclassing with composition is a refactoring issue and is not covered by this ticket, so added a simple test.